src/routers/Home.jsx

```diff
@@ -1,7 +1,22 @@
 import React, { useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
 import NavBar from "../components/NavBar";
+import { useGlobalStore } from "@/common/useGlobalStore.jsx";
+
 
 function HomePage() {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const symmetricKey = useGlobalStore((state) => state.symmetricKey);
+
+  useEffect(() => {
+    if(symmetricKey) {
+      if(!location.state || !location.state.from || location.state.from !== "navbar") {
```

Comment:
When wouldn't we set the location.state or location.state.from? I am also trying to figure out the reason we want to navigate to dashboard in these instances instead of Home?

Reply:
`location.state` doesn't exist if the user went directly to the home page by typing the URL in their browser, but it does if they clicked the "Home" link in the navbar.

I wanted to redirect to Dashboard if the user loads the site while authenticated, but not do the redirect if they've explicitly chosen Home from the navbar.